lib/hpke/src/userinfo_vc.cpp

```diff
@@ -223,6 +223,11 @@ struct UserInfoVC::ParsedCredential
       signature = jws_to_der_sig(signature);
     }
 
+    auto kid = std::optional<std::string>{};
+    if (header.contains("kid")) {
+      kid = header.at("kid").get<std::string>();
+    }
```

Comment:
❓ I will keep reading... but, I don't understand how kid can be optional.  I was under the impression that several public keys from the issuer might be used.  I am not sure that the test client has the ability to handle the case without kid.

Reply:
The `kid` field is OPTIONAL in the [JWK RFC](https://datatracker.ietf.org/doc/html/rfc7517#section-4.5).  There are two plausible modes for a JWT signer to operate in:

1. No `kid`, only one key in the JWK Set 
2. Multiple keys in the JWK Set, use `kid` to distinguish

The specs are silent on which a signer should do, so we should support both options.